modules/transcription-service-microsoft-azure/src/main/java/org/opencastproject/transcription/microsoftazure/MicrosoftAzureTranscriptionService.java

```diff
@@ -520,12 +544,15 @@ void recognizeContinuous(SpeechRecognizer speechRecognizer, String jobId, String
     speechRecognizer.recognized.addEventListener((s, e) -> {
       if (ResultReason.RecognizedSpeech == e.getResult().getReason() && e.getResult().getText().length() > 0) {
         sequenceNumber[0]++;
-        String text = captionFromSpeechRecognitionResult(sequenceNumber[0], e.getResult());
+        List<String> texts = captionFromSpeechRecognitionResult(sequenceNumber[0], e.getResult());
+        sequenceNumber[0] += texts.size() - 1;
```

Comment:
Why increment the value above only to subtract 1 down here?

Reply:
We are never really substracting 1, since `text.size() - 1` is guaranteed to be at least 0 (there will always be at least one result).

`sequenceNumber` is supposed to enumerate the cues in the subtitle file. Previously `captionFromSpeechRecognitionResult(int, SpeechRecognitionResult)` would always return exactly one line, but now it can return any positive number of lines. So to keep `sequenceNumber` correct, the number of additional lines needs to be added.